pkg/controller/history/controller_history.go

```diff
@@ -244,20 +247,25 @@ func (rh *realHistory) CreateControllerRevision(parent metav1.Object, revision *
 	}
 	clone := any.(*apps.ControllerRevision)
 
+	// First attempt to create the revision will not use collisionCount
```

Comment:
To clarify, DaemonSet does still do a retry loop rather than waiting until the next sync. But as part of that loop, on every iteration it calls UpdateStatus to persist the latest collisionCount. It's fine if we drop that UpdateStatus call here because we think it's unnecessary; I was just wondering if you've already thought through the justification for that.

Regarding item #1, do you have a reason for ignoring collisionCount on the first try, even if it's already been set in Status on a previous sync?